docs/tutorials/tensorflow_inference_and_training.md

```diff
@@ -0,0 +1,140 @@
+  - [Epilogue: normal process to run inference](#epilogue-normal-process-to-run-inference)
+- [DeePMD training](#deepmd-training)
+  - [Prologue: install DeePMD-kit and download data](#prologue-install-deepmd-kit-and-download-data)
+  - [Still, all you need are the two lines!](#still-all-you-need-are-the-two-lines)
+  - [Epilogue: normal process to run MD training with DeePMD-kit API](#epilogue-normal-process-to-run-md-training-with-deepmd-kit-api)
+
+
+## BERT inference
+
+BERT models are usually feed with data of dynamic shapes in real production.
+The dynamic shape mainly comes from two aspects, one is varied batch-size, 
+and the other is varied data shape of each sample (e.g., varied sequence lengths).
+The model we show in this tutorial has varied batch-size.
+
+### Prologue: download frozen model
+
+```python
+!mkdir -p model
+!wget -P model http://pai-blade.oss-cn-zhangjiakou.aliyuncs.com/bladedisc_notebook_binaries/models/disc_bert_example/frozen.pb
+```
+
+### All you need are the two lines!
+
+All you need to do to optimize the inference is to add the following two lines of code.
+```python
+import blade_disc_tf as disc
+disc.enable()
+```
+
+### Epilogue: normal process to run inference
+
+After enabling BladeDISC with the two lines above, we can load and run the frozen model with normal process.
+
+First, we load the frozen model and configure the session.
+```python
+import os
+import time
+import numpy as np
+import tensorflow.compat.v1 as tf
+tf.disable_v2_behavior()
+
+graph_def = tf.GraphDef()
+with open('./model/frozen.pb', 'rb') as f:
+    graph_def.ParseFromString(f.read())
+graph = tf.Graph()
+with graph.as_default():
+    tf.import_graph_def(graph_def, name='')
+graph = load_frozen_graph()
+
+session_config = tf.ConfigProto()
+session_config.allow_soft_placement = True
+session_config.gpu_options.allow_growth = True
+# Enable auto-mixed-precision in this example.
+session_config.graph_options.rewrite_options.auto_mixed_precision = 1
+sess = tf.Session(graph = graph, config = session_config)
+```
+
+Finally, we prepare input data and run the session.
+In this example, we fake the input data with varied batch-size to ease the setup.
+```python
+for batch in [2, 2, 4, 1, 1, 8, 8, 2, 16, 2]:
+    feed_dict = {
+        'input_ids_1:0' : np.ones((batch, 384), dtype=int),
+        'segment_ids_1:0' : np.zeros((batch, 384), dtype=int),
+        'input_mask_1:0' : np.ones((batch, 384), dtype=int),
+    }
+    outs = sess.run(fetch, feed_dict = feed_dict)
+```
+
+The above code shows the complete process to optimize BERT inference model with BladeDISC.
+Please refer to [TensorFlow BERT Inference Example](/examples/TensorFlow/Inference/BERT) 
+for more scripts to compare the performance of BladeDISC optimization with naive TensorFlow and XLA.
+
+
+## DeePMD training
+
+We take a deep learning based model for molecular dynamics (MD) to show how to optimize a training model with BladeDISC.
```

Comment:
The length of each line should be limited for a md file. You can use a 'rewrap' plugin of Vscode to reformat the file.

Reply:
Done. Thanks.